posts/2020/03/pytorch-tensoriterator-internals.md

```diff
@@ -0,0 +1,236 @@
+<!--
+.. title: PyTorch TensorIterator Internals
+.. slug: pytorch-tensoriterator-internals
+.. date: 2020-03-12 22:39:56 UTC-05:00
+.. tags: 
+.. category: 
+.. link: 
+.. description: 
+.. type: text
+-->
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-generate-toc again -->
+**Table of Contents**
+
+- [Introduction](#introduction)
+- [History of TensorIterator](#history-of-tensoriterator)
+    - [TH iterators](#th-iterators)
+    - [Limitations of TH iterators](#limitations-of-th-iterators)
+- [Basics of TensorIterator](#basics-of-tensoriterator)
+- [Performing iterations](#performing-iterations)
+    - [Iteration details](#iteration-details)
+        - [Setting tensor iteration dimensions](#setting-tensor-iteration-dimensions)
+- [Conclusion](#conclusion)
+
+<!-- markdown-toc end -->
+
+# Introduction
+
```

Comment:
A first sentence could be a more general intro. For example:
_PyTorch is one of the leading frameworks for deep learning. Its core data structure is `Tensor`, a multi-dimensional array implementation with many advanced features like auto-differentiation._